File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman_map = {"I" : 1, "V" : 5, "X" : 10, "L" : 50, "C" : 100, "D" : 500, "M" : 1000}
        total = 0
        prev = 0
        for letter in s:
            if roman_map[letter] > prev:
                total -= prev
                total += roman_map[letter] - prev
            else:
                total += roman_map[letter]
                
            prev = roman_map[letter]
            
        return total
            
            
        

File: 0013-roman-to-integer/0013-roman-to-integer.cs
//using System.Collections.Generic;
public class Solution {
    public int RomanToInt(string s) {
        
        Dictionary<char, int> romanMap = new Dictionary<char, int>
{
    { 'I', 1 },
    { 'V', 5 },
    { 'X', 10 },
    { 'L', 50 },
    { 'C', 100 },
    { 'D', 500 },
    { 'M', 1000 }
};
        int prev = 0;
        int total = 0;
        
        for(int i = 0; i < s.Length; i++)
        {
            if(romanMap[s[i]] > prev)
            {
                total -= prev;
                total += romanMap[s[i]] - prev;
            }
            else
            {
                total += romanMap[s[i]];
            }
            prev = romanMap[s[i]];
        }
        return total;
        
    }
}